artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPSessionCallback.java

```diff
@@ -496,15 +499,21 @@ public void serverSend(ProtonServerReceiverContext context,
             }
          } else {
             message.setConnectionID(receiver.getSession().getConnection().getRemoteContainer());
-            // We need to transfer IO execution to a different thread otherwise we may deadlock netty loop
-            sessionExecutor.execute(() -> inSessionSend(context, transaction, message, delivery, receiver, routingContext));
+            // If there are pending messages (durable) on the executors, we should still use the executors
+            // otherwise we risk running out of order
+            if (pendingSettles == 0 && !message.isDurable()) {
```

Comment:
This should fix your test... as this will guarantee no execution for non durable.